### ISSUE TITLE
How Are sig_inter_.pkl and train_inter_*.txt Generated in SVDF?

### ISSUE THREAD

**Issue by xiaoxiaoguaiyale** [hh22814/SVDF#5]
Hello, and thank you very much for releasing the SVDF code and dataset.

I am currently reproducing and studying SVDF in detail, and I have a few questions that I could not fully resolve from the paper or the repository. I would be very grateful for your guidance.

About sig_inter_.pkl
How were these .pkl files generated?
Were they derived directly from PBSIM-simulated BAM files?
If possible, could you share the general workflow (PBSIM BAM → signatures → pkl)?

About labels in train_inter_*.txt
The label column shows DEL / INS / None.
Are these labels obtained by matching PBSIM’s VCF, GIAB truth, or mainly generated from geometric rules (distance_on_read, distance_on_reference, orientation, etc.)?

About the inter-RF training data
For training the random forest model (e.g., model_sim_indel_rf_test_.pkl),
did you use both sig_inter_.pkl and train_inter_*.txt?
Or is one of them the primary supervised dataset?

Extending SVDF to new simulated data (e.g., VISOR)
If generating new simulation data, what is the recommended way to obtain:
BAM → SV signatures → labels → RF training dataset?

Thank you very much for your time and for sharing this work.
Any small clarification would greatly help my reproduction of SVDF.

**Comment by coopsor** [comment]
1.The signature objects were obtained by running SVDF in sensitive mode on the simulated dataset. They were not generated by any additional scripts. You may refer to the “Implementation of training dataset and simulated sample generation” section in the Supplementary Notes of our paper for the overall workflow.

2.The labels were assigned by matching each signature to the GIAB benchmark SVs based on their genomic position and length.

3.The random forest model was trained using all signatures from chromosomes 1–11 of both the simulated dataset and the GIAB dataset. The simulated data were mainly introduced to supply SV types other than indels, since GIAB contains only indels.

4. After extracting signatures with SVDF in sensitive mode, we matched each signature with the GIAB benchmark set using both position and length. If a match was found, the signature was labeled with the corresponding SV type; otherwise, it was labeled as None.

All of these details can be found in the supplementary materials of our [paper.](https://doi.org/10.1093/bib/bbae336)

Please feel free to let me know if any further clarification is needed.

**Comment by xiaoxiaoguaiyale** [comment]
Thank you very much for your quick and detailed reply!

**Comment by xiaoxiaoguaiyale** [comment]
我想进一步确认 SVDF 中 inter-alignment signature 的标签生成流程，以及随机森林模型的训练方式。

1. inter-alignment signature 的标签是否是通过将 sensitive 模式输出与 SURVIVOR 生成的 BED 真值进行匹配得到的？
根据论文描述，我目前的理解是：

对 PBSIM 模拟的 BAM 使用 SVDF 的 sensitive 模式，提取所有 inter-alignment signatures；

SURVIVOR 在模拟时生成 BED 格式 的真值 SV；

通过 signature 的位置和长度与该 BED 真值匹配，从而得到 sig_inter_.pkl 中每个 signature 的标签。

请问这是否是完整且正确的标注流程？

2. 开源的 RF 训练脚本只包含 DEL / INS / None 三类，与 general 模式所使用的五分类模型不一致，二者之间的关系是什么？

在 rf_model.py 中，标签映射如下：

#将最后一列中DEL,INS,None分别替换为0,1,2
for df in [df_indel, df_indel_2, df_indel_3]:
    df['label'] = df['label'].replace('DEL', 0)
    df['label'] = df['label'].replace('INS', 1)
    df['label'] = df['label'].replace('None', 5)

这看起来只训练了一个 三分类（主要用于 indel） 的模型，但在 general 模式中，SVDF 需要预测：

DEL、INS、DUP、INV、TRA（五分类）

**Comment by coopsor** [comment]
> 我想进一步确认 SVDF 中 inter-alignment signature 的标签生成流程，以及随机森林模型的训练方式。
> 
> 1. inter-alignment signature 的标签是否是通过将 sensitive 模式输出与 SURVIVOR 生成的 BED 真值进行匹配得到的？
>    根据论文描述，我目前的理解是：
> 
> 对 PBSIM 模拟的 BAM 使用 SVDF 的 sensitive 模式，提取所有 inter-alignment signatures；
> 
> SURVIVOR 在模拟时生成 BED 格式 的真值 SV；
> 
> 通过 signature 的位置和长度与该 BED 真值匹配，从而得到 sig_inter_.pkl 中每个 signature 的标签。
> 
> 请问这是否是完整且正确的标注流程？
> 
> 2. 开源的 RF 训练脚本只包含 DEL / INS / None 三类，与 general 模式所使用的五分类模型不一致，二者之间的关系是什么？
> 
> 在 rf_model.py 中，标签映射如下：
> 
> #将最后一列中DEL,INS,None分别替换为0,1,2 for df in [df_indel, df_indel_2, df_indel_3]: df['label'] = df['label'].replace('DEL', 0) df['label'] = df['label'].replace('INS', 1) df['label'] = df['label'].replace('None', 5)
> 
> 这看起来只训练了一个 三分类（主要用于 indel） 的模型，但在 general 模式中，SVDF 需要预测：
> 
> DEL、INS、DUP、INV、TRA（五分类）

1.你理解的完全正确。
2.这一步标签映射只是针对来自GIAB数据集的indel标签进行预处理，目的是为了将GIAB数据集的indel两类与模拟数据集的其他三类标签合并进行统一训练。包含完整的标签数据在df_ = pd.concat([pd.DataFrame(data), df_indel, df_indel_2, df_indel_3])这一行代码中

**Comment by xiaoxiaoguaiyale** [comment]
非常感谢您的帮助
